File: tools/pilot-reader/tests/wire_fixture.py
"""Shared final-artifact reader for Python assertions, old + compact formats.

Actual production JS decode/boot is separately exercised by the strict DOM suites.
This independent adapter restores the same projection; never changes raw records.
"""
import json
from entry_fixture import Tree


def elements(tree):
    yield tree
    for child in tree['children']:
        yield from elements(child)


def decode_html(html):
    parser = Tree(); parser.feed(html)
    by = {e['attrs']['id']:e for e in elements(parser.root) if 'id' in e['attrs']}
    data = json.loads(by['graph-data']['text'])
    if 'wireVersion' not in data:
        return data
    assert data['wireVersion'] == 1
    assert 'records' not in data
    data['records'] = json.loads(by['canonical-records']['text'])
    for n in data['graph']['nodes']:
        n['raw'] = data['records'][n['id']]
        n['title'] = n['raw']['title']; n['kind'] = n['raw']['kind']
        n.setdefault('label', n['title'])
    for e in data['graph']['edges']:
        e['raw'] = data['records'][e['id']]
    if 'projectId' in data:
        data['project'] = data['records'][data['projectId']]
    return data

File: tools/pilot-reader/tests/wire-fixture.js
'use strict';
// Shared test adapter: current artifacts use the actual production decoder;
// immutable historical artifacts retain their own embedded old contract.
const {decodeWire}=require('../bootstrap.js');
function readData(doc) {
  const wire=JSON.parse(doc.getElementById('graph-data').textContent);
  if(wire.wireVersion===undefined)return wire;
  const decoded=decodeWire(doc);
  // Phase B decodeWire returns {data, records}; tests expect the data object.
  return decoded.data!==undefined && decoded.records!==undefined ? decoded.data : decoded;
}
function writeData(doc,data) {
  if(data.wireVersion===undefined) {
    doc.getElementById('graph-data').textContent=JSON.stringify(data);return;
  }
  const wire={...data,graph:{...data.graph}};delete wire.records;delete wire.project;
  wire.graph.nodes=data.graph.nodes.map(n=>{
    const item={...n};delete item.raw;
    if(item.title===n.raw.title)delete item.title;
    if(item.kind===n.raw.kind)delete item.kind;
    return item;
  });
  wire.graph.edges=data.graph.edges.map(e=>{const item={...e};delete item.raw;return item;});
  doc.getElementById('graph-data').textContent=JSON.stringify(wire);
  doc.getElementById('canonical-records').textContent=JSON.stringify(data.records);
}
module.exports={readData,writeData};
